### PR TITLE
feat: use new proxy 🥳 fix #1396

### LINKED ISSUE #1396: Response message get processed by JSON.stringify() multiple times.
### What happens?

Response message get processed by JSON.stringify() multiple times.

See example
Screenshot 2024-04-08 233406 (2)

Method "ping" actually returns string `"Pong\n"` not the `"\\"Pong\\\\n\\""`.

I want to note that i'm not using data-proxy-url.
It seems that using a proxy may change the visible response value.

### What did you expect to happen?

Raw data that server returns:
Screenshot 2024-04-08 233406 (3)

### How can we reproduce the issue?

The same result can be achieved by using this command:
`console.log(JSON.stringify(JSON.stringify("Pong\n")));`

### Swagger/OpenAPI Example

_No response_

### PR DESCRIPTION
**Pull request by marclave**
This PR migrates to the new proxy, which is now hosted on https://proxy.scalar.com

We used to have a custom proxy, which returned JSON always. The new proxy is a raw HTTP proxy (written in Go), which expects to get a `?scalar_url=` query parameter, everything else (headers and such) aren’t touched at all and just proxied through.

If you’re using the old proxy (https://api.scalar.com/request-proxy) we [magically replace it](https://github.com/scalar/scalar/pull/1648) with the new URL (https://proxy.scalar.com) on the fly. You’ll receive a deprecation warning, though. Please update your configuration to use the new proxy URL.

If you used to self-host our `@scalar/api-client-proxy` we kindly ask you to switch to our new proxy (examples/proxy-server) written in Go. If you need help with that, reach out to us on Discord.

**Additional improvements**
* the proxy is skipped for requests to localhost, 127.0.0.1 and ::1
* Network errors (eg. domain doesn’t exist) are shown
* Not-JSON content is correctly formatted again
* CORS errors at least show "ERR_NETWORK" in the response body
* some more tests
* more lines deleted than added

**Additional fixes**
* #1396
* Since #1302 variables in URLs were not replaced anymore, this PR fixes it

**Preview**

Screenshot 2024-05-15 at 16 11 58
Screenshot 2024-05-15 at 16 12 14
Screenshot 2024-05-15 at 16 25 23

(Comment hijacked by @hanspagel)

### CODE REVIEW

File: packages/api-client/src/helpers/sendRequest.ts [hunk 1]
```diff
@@ -103,18 +103,14 @@ export async function sendRequest(
...
-        url: proxyUrl,
-        data: requestConfig,
-      }
-    : {
-        method: requestConfig.method,
-        url: requestConfig.url,
-        headers: requestConfig.headers,
-        data: requestConfig.data,
-      }
+  const axiosRequestConfig: AxiosRequestConfig = {
```
Comment (marclave) [inline packages/api-client/src/helpers/sendRequest.ts:106]: @hanspagel I think we need to handle the response stuff here ✨ and the check if using old proxy

File: packages/api-client/src/helpers/sendRequest.ts [hunk 2]
```diff
@@ -130,14 +126,6 @@ export async function sendRequest(
 
   const response: ClientResponse = await axios(axiosRequestConfig)
     .then((result) => {
-      // With proxy
```
Comment (marclave) [inline packages/api-client/src/helpers/sendRequest.ts:133]: @hanspagel not sure if we need to do anything here if using the old proxy

**Comment by hanspagel** [comment]
Which one is “Hans' old PR”? 😳

And we need to have tests for all scenarios. We need to somehow spin up the go proxy in the test environment. 🤔

**Review by hanspagel — APPROVED** [review]
LGTM, wdyt @marclave and others?

File: packages/api-client/src/components/ApiClient/Response/ResponseBody.vue [hunk 3]
```diff
@@ -1,39 +1,42 @@
...
+const props = defineProps<{
+  response?: any
+}>()
 
+// In order to render the response body, we need to know the media type.
 const mediaType = computed(() => {
-  const contentTypeHeader = props.headers.find(
-    (header) => header.name.toLowerCase() === 'content-type',
+  // Transform all header keys to lowercase
+  const headers = Object.fromEntries(
```
Comment (geoffgscott) [inline packages/api-client/src/components/ApiClient/Response/ResponseBody.vue:17]: Would be nice to see a `normalizeHeaders` utility function here that we can share.

**Comment by amritk** [comment]
The code looks fine, but I tried using the preview and it says 201 but I got a CORS error? No idea whats going on lol
image
image

**Review by amritk — APPROVED** [review]
Tested locally and works great! The issue is with the deploy preview

**Comment by hanspagel** [comment]
The cors header has *, * as a value! I fixed this in the Go proxy, but proxy.scalar.com doesn’t have the fix yet. Because it’s in this PR here. 😂

**Review by marclave — COMMENTED** [review]
HUGE! let's goooo

**Pull request merged**